rest/data/project.go

```diff
@@ -417,3 +448,33 @@ func (pc *MockProjectConnector) UpdateProjectRevision(projectID, revision string
 func (ac *MockProjectConnector) GetVersionsInProject(project, requester string, limit, startOrder int) ([]restModel.APIVersion, error) {
 	return nil, nil
 }
+
+func (pc *MockProjectConnector) GetProjectSettingsEvent(p *model.ProjectRef) (*model.ProjectSettingsEvent, error) {
```

Comment:
The purpose of a mock is to avoid using the database (see above, for example, how FindProjectById iterates through p.CachedProject). Either you want to write a test that explicitly uses the DBProjectConnector so you don't have to implement the mock (I'm assuming failed tests were why you did it), or you should incorporate this into the Cached parameters. Let me know if you have questions about that.